model/task/task.go

```diff
@@ -2586,12 +2586,59 @@ func GetTasksByVersion(versionID, sortBy string, statuses []string, variant stri
 		addDisplayStatus,
 	}
 	if len(statuses) > 0 {
-		pipeline = append(pipeline, bson.M{
-			"$match": bson.M{
-				DisplayStatusKey: bson.M{"$in": statuses},
-			},
-		})
+		// statuses can include a status (i.e. aborted) that is not valid for matching on the DisplayStatusKey
```

Comment:
This works for filtering, but not for sorting. If you add a branch in the addDisplayStatus logic to return "aborted" for tasks that should show that in their status label, I think that'll take care of both, and you don't need the extra logic here or in the filter for base statuses

Reply:
oooooooh!!!!